src/Compilers/VisualBasic/Portable/Symbols/Wrapped/WrappedParameterSymbol.vb

```diff
@@ -153,6 +153,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             End Get
         End Property
 
+        Friend Overrides ReadOnly Property CallerArgumentExpressionParameterIndex As Integer
+            Get
+                Return Me._underlyingParameter.CallerArgumentExpressionParameterIndex
```

Comment:
>Return Me._underlyingParameter.CallerArgumentExpressionParameterIndex [](http://example.com/codeflow?start=16&length=69)

It is hard to confirm if this implementation is correct for existing derived types and will be correct for future derived types. I think we should let derived types to handle implementation. For C# as well.

Reply:
The docs on WrappedParameterSymbol says the following (in both C# and VB):

```
    ''' <summary>
    ''' Represents a parameter that is based on another parameter.
    ''' When inheriting from this class, one shouldn't assume that 
    ''' the default behavior it has is appropriate for every case.
    ''' That behavior should be carefully reviewed and derived type
    ''' should override behavior as appropriate.
    ''' </summary>
```

But let me know if you still prefer to move it to the derived types (in this case I'll also move all the 'caller info'-related properties).

Reply:
> But let me know if you still prefer to move it to the derived types (in this case I'll also move all the 'caller info'-related properties).

Yes, the comment doesn't change anything. We learned on multiple occasions that it is not realistic to expect that every dev would review all APIs that can be overridden by a new derived type. That is why we adopted a strategy that virtual APIs are not added to symbols, only abstract are added, implementation must go to a leaf types (most derived). This way, a dev is forced to think what implementation should be for a new type. We can accept some deviations for very specialized hierarchies, like SubstitutedParameterSymbol, where we can upfront predict what correct implementation must be for all types derived from it. With this type, I do not believe we can make any assumption.